lua/seraphimunits.lua

```diff
@@ -595,18 +618,20 @@ SEnergyBallUnit = ClassUnit(SHoverLandUnit) {
                     end
                 end
 
+                ---@type Unit
                 local target = table.random(filteredUnits)
+
                 if target then
```

Comment:
```suggestion
                if target and (not IsAlly(self.Army, target.Army) or Random(1,5) <= 1) then
```
This could moderate ally damage while retaining visual effects. Some experimentation would be needed to determine the most balanced chance to hit allies. A storm has 152000 potential damage (2000 dmg, 4 tick waits, 30 second lifetime). I did 9 tests of 3 ythotha standing next to each other in a row, and killing the one on the right got me an average of 28700 damage to allies per storm, so for this suggestion I'll use a 1/5 chance.
In addition, there is currently an issue that the storm mostly only damages allies right next to the minimum range, so to retain that feature you could reduce the chance as range increase, although I don't think that's a good mechanic (it's hard to notice as a player), so I didn't try.